### PR TITLE
Development

### PR DESCRIPTION
**Pull request by AnkitKarnAK**
# RacketStream Video Lbrary web app

## Features:
1. Auth (Login & Signup)
2. Private Routes
3. Video Page
4. Liked Videos
5. Watch Later Videos
5. Playlist Management (Create or Delete User Playlists)
6. Add/Remove videos to/from playlist

### CODE REVIEW

**Comment by Akshaay9** [comment]
Hey Ankit,
here`s few of my suggestion

Feature:-
1. Try to add features of `Like video and add to playlist` on individual videos too`
2. I am able to create multiple playlists with same name, so have a check on that in backend or in frontend

Code:-
1. On `api-request.js` file, u could have added one more case inside switch for POST,and used that for making all `get` and `post `request intead of creating multiple functions like `getVideoFromServer` etc:-
2. In AuthContext, instead of creating multiple useStates, u could have used useReducer and an initialState to do perform and store all the necessary user details
3. (suggestion) try to write functions like `signupNewUser` and `signupNewUser` outside of your contextAPI, so that code would be easy to scale same thing goes for data-reducer.js.
4. On login page, it would be great if u could suggest what is the required password format, for users

Apart from that, everything looks Good.!
Akshay.

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,112 @@
...
+        dispatch({ type: "SET_VIDEOS", payload: response.data.videos });
+      } catch (err) {
+        alert("failed to fetch ", err);
+        console.error(err);
+      }
+    })();
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (isUserLogin) {
```
Comment (uditkumar01) [inline src/App.js:47]: keep this function in a new file and import it from there

File: src/components/WatchLaterItem.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            </div>
+          </Link>
+          <div
+            className={
+              isAlreadyAdded2(watchLaterVideos, watchLaterItem._id)
+                ? "card-badge-top-right in-watch-later"
+                : "card-badge-top-right"
+            }
+            onClick={() => {
+              (async () => {
```
Comment (uditkumar01) [inline src/components/WatchLaterItem.js:50]: make it a separate function

File: src/components/pages/Signup.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
+import React, { useState } from "react";
+import Loader from "react-loader-spinner";
+
+import { useAuthContext } from "../../context/auth-context";
+import { Link, Navigate } from "react-router-dom";
+
+const Signup = () => {
+  const { signupNewUser, isUserLogin } = useAuthContext();
+
+  const [userEmail, setUserEmail] = useState("");
```
Comment (uditkumar01) [inline src/components/pages/Signup.js:10]: u can use useReducer here as it has a lot of useEffect hooks

File: src/components/pages/Signup.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      return;
+    } else {
+      setServerError(
+        res?.data?.errorMessage || "Something went wrong, please try again!"
+      );
+      setLoading(false);
+    }
+  };
+
+  const emailValidator = () => {
```
Comment (uditkumar01) [inline src/components/pages/Signup.js:39]: put these functions in separate utility file as these are generally used on both login and sign up components

**Review by uditkumar01 — CHANGES_REQUESTED** [review]

**Pull request merged**